packages/gallery/src/components/item/itemHover.js

```diff
@@ -54,17 +54,107 @@ export default class ItemHover extends GalleryComponent {
     return true;
   }
 
+  getOverlayStyle() {
+    const { styleParams } = this.props;
+    const { overlayPosition } = styleParams;
+    const isHorizontal =
+      overlayPosition === GALLERY_CONSTS.overlayPositions.LEFT ||
+      overlayPosition === GALLERY_CONSTS.overlayPositions.RIGHT ||
+      overlayPosition === GALLERY_CONSTS.overlayPositions.CENTERED_HORIZONTALLY;
+    if (isHorizontal) {
+      return this.calculateHorizontalOverlayStyles();
+    }
+    return this.calculateVerticalOverlayStyles();
+  }
+
+  calculateVerticalOverlayStyles() {
+    const { styleParams, imageDimensions } = this.props;
+    const { overlayPosition, overlaySize, overlaySizeType, overlayPadding } =
+      styleParams;
+    let { width, height, marginTop, marginLeft } = imageDimensions;
+    let style = { maxHeight: height, maxWidth: width };
+    let overlayHeight =
+      (overlaySizeType === 'PERCENT'
+        ? height * (overlaySize / 100)
+        : overlaySize) -
+      2 * overlayPadding;
+    let overlayWidth = width - 2 * overlayPadding;
+    let marginRight = overlayPadding;
+    let marginBottom = overlayPadding;
+    marginTop += overlayPadding;
```

Comment:
New code should be written [without mutations](https://github.com/immutable-js/immutable-js/#introduction).  This is out of scope for our meeting but still important